storage/statedb/cache_redis_test.go

```diff
@@ -98,22 +99,66 @@ func TestNewRedisCache(t *testing.T) {
 	assert.Equal(t, bytes.Compare(value, hasValue), 0)
 }
 
-// TestNewRedisCache_Set_LargeData check whether redis cache can store an large data (5MB).
-func TestNewRedisCache_Set_LargeData(t *testing.T) {
+// TestRedisCache_Set_LargeData check whether redis cache can store an large data (5MB).
+func TestRedisCache_Set_LargeData(t *testing.T) {
 	cache, err := NewRedisCache(getTestRedisConfig())
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	key, value := randBytes(32), randBytes(5*1024*1024) // 5MB value
 	cache.Set(key, value)
+	time.Sleep(100 * time.Millisecond)
 
 	retValue := cache.Get(key)
 	assert.Equal(t, bytes.Compare(value, retValue), 0)
 }
 
-// TestNewRedisCache_Timeout test timout feature of redis client.
-func TestNewRedisCache_Timeout(t *testing.T) {
+func TestRedisCache_Set_LargeNumberItems(t *testing.T) {
+	cache, err := NewRedisCache(getTestRedisConfig())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	itemsLen := redisSetItemChannelSize * 2
+	items := make([]setItem, itemsLen)
+	for i := 0; i < itemsLen; i++ {
+		items[i].key = randBytes(32)
+		items[i].value = randBytes(500)
+	}
+
+	// successCh returns struct{} when the following goroutine checks all items are set in the redis
+	successCh := make(chan struct{})
+	go func() {
+		for i := 0; i < itemsLen; i++ {
+			v := cache.Get(items[i].key)
+			if !bytes.Equal(v, items[i].value) {
+				// if the item is not set yet, wait and retry
+				time.Sleep(100 * time.Millisecond)
+				i--
+			}
+		}
+		successCh <- struct{}{}
+	}()
+
+	for i := 0; i < itemsLen; i++ {
+		if i == redisSetItemChannelSize {
+			// sleep for a while because Set command can drop an item if setItem channel is full
+			time.Sleep(2 * time.Second)
```

Comment:
Is there a reason you this time to 2 seconds? It is a long time for a test.

Reply:
Just a heuristic number derived in my test. I gave enough time for the sleep because it only happens once. We could decrease it more tightly also. Do you have any suggestions?